File: ide/vstudio/ALox-CS-.Net45-Test-Release-Log/SimpleTests.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using com.aworx.util;
using com.aworx.lox;
using com.aworx.lox.loggers;
using com.aworx.lox.tools;

namespace ALox_CS_Test_Perf	 {

class SimpleTests
{
	// #################################################################################################
	// static entrance (Main)
	// #################################################################################################
	static void Main( string[] args )
	{
		// create us
		SimpleTests test= new SimpleTests();
		
		// do some release logging tests. 
		Console.WriteLine( "PRINT: Release logging test:" );
			test.testReleaseLogging();
		Log.Reset();

		// do some performance tests. 
		Console.WriteLine( "PRINT: Performance test:" );
			test.performanceTest();
		Log.Reset();
		
		Console.WriteLine( "PRINT: Thats it!" );

	}

	// #################################################################################################
	// Fields
	// #################################################################################################
	ConsoleLogger cl;
	MemoryLogger  ml;

	// #################################################################################################
	// Test functions
	// #################################################################################################
	
	void performanceTest()
	{
		cl=	new ConsoleLogger();
		#if ALOX_DEBUG || ALOX_REL_LOG
			cl.EnableAppConsole=		true;
			cl.EnableVSDebugConsole=	false;
		#endif
		ml= new MemoryLogger();

		Log.AddLogger( cl, Log.DomainLevel.WarningsAndErrors );
		Log.AddLogger( ml, Log.DomainLevel.Off );

		Log.RegDomain( "CON", Log.Scope.SourceFile );
		Log.RegDomain( "MEM", Log.Scope.None  );
		Log.RegDomain( "BOTH",Log.Scope.None );

		Log.SetDomain( "CON", Log.DomainLevel.All, true, "Console" );
		Log.SetDomain( "MEM", Log.DomainLevel.Off, true, "Console" );
		Log.SetDomain( "MEM", Log.DomainLevel.All, true, "Memory" );
		Log.SetDomain( "CON", Log.DomainLevel.Off, true, "Memory" );
		Log.SetDomain( "BOTH",Log.DomainLevel.All, true  );


		Log.Info( "Logging simple info lines" );

		MString	msgBuf=		new MString( );
		long	fastest=	long.MaxValue;
		for ( int i= 0 ; i < 500 ; i++ )
		{
			#if ALOX_DEBUG || ALOX_REL_LOG
				ml.Buffer.Clear();
			#endif

			long t= Ticker.Now();
				simpleInfoLines( 1000 );
			t= Ticker.Now() - t;

			if ( fastest > t )
			{
				fastest= t;
				Log.Line( Log.Level.Info, msgBuf.Clear().Append( "Pass " ).Append( i, 3).Append( " is new fastest:  ")
														.Append( (int) Ticker.ToMillis( fastest ), 0).Append( " millis (").Append( (int) fastest ).Append( " ticks) per 1000 logs.") );
			}
		}

		Log.Line( Log.Level.Info, msgBuf.Clear().Append( "Fastest " ).Append( (int) Ticker.ToMillis( fastest ), 0).Append( " millis (").Append( (int) fastest ).Append( " ticks) per 1000 logs.") );
	}

	void simpleInfoLines( int qty )
	{
		for ( int i= 0 ; i < qty ; i++ )
			Log.Info( "MEM", "Test Line" );
	}

	public void testReleaseLogging()
	{
		// create a lox for release logging
		Lox				lox= 	new Lox();
		
		// add and configure a memory logger
		MemoryLogger	ml=			new MemoryLogger();
		#if ALOX_DEBUG || ALOX_REL_LOG
			ml.LineFormatter.Format= 	new MString( "[%TE]%L[%O]: ");
		#endif
		lox.AddLogger( ml, 	Log.DomainLevel.All );
		
		// We do not work with default domains, as we will obfuscate the code.
		// Obfuscated code and release logging does not allow default domains, because
		// the rely on caller information and reasonable package, class and method names
		lox.RegDomain( "RelLog",	Log.Scope.None );
		lox.SetDomain( "RelLog",	Log.DomainLevel.All  );

		// all log invocations provide the "domain" parameter explicitly to be safe when
		// obfuscation is enabled!
		lox.Info( "RelLog", "Hello ALox, thank you for providing release logging!" );
		lox.Info( "RelLog", "Let's see if LogTools is available." );
		lox.Info( "RelLog", "We need to provide our lox as a parameter to all LogTools methods!" );
		
		// instance() will not give very nice member names, if obfuscated!
		LogTools.Instance( "RelLog", Log.Level.Info, lox, 2, "Logging the Lox:", 0, lox );
		
		// also, the exception stack trace should be not too easy to read. Proguard provides a tool "retrace"...
		lox.Info( "RelLog", "Logging exceptions: " );
		LogTools.Exception( "RelLog", Log.Level.Error, new Exception("This is not a real Exception", new Exception("...unreal inner")), " Logging instance 'this':", 0, lox );

		lox.Info( "RelLog", "That's it for now. More release logging tests to come...stay tuned!" );

		// copy the memory logger's output to the console
		if ( ml.Buffer.Length > 0 ) 
			Console.WriteLine(  ml.Buffer.ToString() );
		else
			Console.WriteLine(  "  Even release logging got pruned - nothing logged!" );
	}


} // class
}

File: ide/eclipse.kepler/ALox-Java-1.6-Test/src/tests/com/aworx/lox/SimpleTests.java
package tests.com.aworx.lox;


import com.aworx.util.*;
import com.aworx.lox.*;
import com.aworx.lox.loggers.ConsoleLogger;
import com.aworx.lox.loggers.MemoryLogger;
import com.aworx.lox.tools.LogTools;


public class SimpleTests
{


	// #################################################################################################
	// static entrance (Main)
	// #################################################################################################
	/**
	 *  In-between the different tests, we have to do some System.out.println() here to have at least 
	 *  a minimum output after pruning ALox (as this software does not do anything but logging).
	 *  
	 * @param args
	 */
	public static void main(String[] args)
	{
	
		// create us
		SimpleTests test= new SimpleTests();
		
		// do some release logging tests. 
		System.out.println( "PRINT: Release logging test:" );
			test.testReleaseLogging();
		Log.reset();

		// do some performance tests. 
		System.out.println( "PRINT: Performance test:" );
			test.performanceTest();
		Log.reset();
		
		System.out.println( "PRINT: Thats it!" );
	}

	// #################################################################################################
	// Fields
	// #################################################################################################

	// #################################################################################################
	// Test functions
	// #################################################################################################
	
	void performanceTest()
	{
		ConsoleLogger cl=	new ConsoleLogger( "Console" ); 
		MemoryLogger  ml= new MemoryLogger( "Memory");

		Log.addLogger( cl, Log.DomainLevel.WARNINGS_AND_ERRORS );
		Log.addLogger( ml, Log.DomainLevel.OFF );

		Log.regDomain( "CON", Log.Scope.METHOD );
		
		Log.regDomain( "MEM", Log.Scope.NONE  );
		Log.regDomain( "BOTH",Log.Scope.NONE );

		Log.setDomain( "CON", Log.DomainLevel.ALL, true, "Console" );
		Log.setDomain( "MEM", Log.DomainLevel.OFF, true, "Console" );
		Log.setDomain( "MEM", Log.DomainLevel.ALL, true, "Memory" );
		Log.setDomain( "CON", Log.DomainLevel.OFF, true, "Memory" );
		Log.setDomain( "BOTH",Log.DomainLevel.ALL, true  );

		//Log.LogConfig( "CON", Log.Level.Info, "Log Config follows..." );
		Log.info( "Logging simple info lines" );
		long	fastest= 	Long.MAX_VALUE;
		for ( int i= 0 ; i < 500 ; i++ )
		{
			ml.buffer.clear();
			long t= Ticker.now();
				simpleInfoLines( 1000 );
			t= Ticker.now() - t;
			if ( fastest > t )
			{
				fastest= t;
				Log.line( "CON", Log.Level.INFO, Log.buf()	.append( "Pass " ).append( i, 3).append( " is new fastest:  ")
															.append( (int) Ticker.toMillis( fastest ), 0).append( " millis (").append( (int) fastest ).append( " ticks) per 1000 logs.") );
			}

		}

		Log.line( Log.Level.INFO, Log.buf()	.append( "Fastest " ).append( (int) Ticker.toMillis( fastest ), 0).append( " millis (").append( (int) fastest ).append( " ticks)  per 1000 logs.") );
	}

	static void simpleInfoLines( int qty )
	{
		for ( int i= 0 ; i < qty ; i++ )
			Log.info( "MEM", "Test Line" );
	}


	public void testReleaseLogging()
	{
		// create a lox for release logging
		Lox				lox= 	new Lox();
		
		// add and configure a memory logger
		MemoryLogger	ml=			new MemoryLogger();
		ml.lineFormatter.format= 	new MString( "[%TE]%L[%O]: ");
		lox.addLogger( ml, 	Log.DomainLevel.ALL );
		
		// We do not work with default domains, as we will obfuscate the code.
		// Obfuscated code and release logging does not allow default domains, because
		// the rely on caller information and reasonable package, class and method names
		lox.regDomain( "RelLog",	Log.Scope.NONE );
		lox.setDomain( "RelLog",	Log.DomainLevel.ALL  );

		// all log invocations provide the "domain" parameter explicitly to be safe when
		// obfuscation is enabled!
		lox.info( "RelLog", "Hello ALox, thank you for providing release logging!" );
		lox.info( "RelLog", "Let's see if LogTools is available." );
		lox.info( "RelLog", "We need to provide our lox as a parameter to all LogTools methods!" );
		
		// instance() will not give very nice member names, if obfuscated!
		LogTools.instance( "RelLog", Log.Level.INFO, lox, 2, "Logging the Lox:", 0, lox );
		
		// also, the exception stack trace should be not too easy to read. Proguard provides a tool "retrace"...
		lox.info( "RelLog", "Logging exceptions: " );
		LogTools.exception( "RelLog", Log.Level.ERROR, new Exception("This is not a real Exception", new Exception("...unreal inner")), " Logging instance 'this':", 0, lox );

		lox.info( "RelLog", "That's it for now. More release logging tests to come...stay tuned!" );

		// copy the memory logger's output to the console
		if ( ml.buffer.length > 0 ) 
			System.out.println(  ml.buffer.toString() );
		else
			System.out.println(  "  Even release logging got pruned - nothing logged!" );
	}

}
